source/tanya/math/random.d

```diff
@@ -160,6 +160,181 @@ version (linux)
+
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379886(v=vs.85).aspx
+        // For performance reasons, we recommend that you set the pszContainer
+        // parameter to NULL and the dwFlags parameter to CRYPT_VERIFYCONTEXT
+        // in all situations where you do not require a persisted key.
+        // CRYPT_SILENT is intended for use with applications for which the UI cannot be displayed by the CSP.
+        if (!CryptAcquireContextW(&hProvider, null, null, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT))
+        {
+            if (GetLastError() == NTE_BAD_KEYSET)
+            {
+                // Attempt to create default container
+                if (!CryptAcquireContextA(&hProvider, null, null, PROV_RSA_FULL, CRYPT_NEWKEYSET | CRYPT_SILENT))
+                    return false;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    class PlatformEntropySource : EntropySource
+    {
+        private HCRYPTPROV hProvider;
+
+        /**
+         * Uses CryptGenRandom.
+         */
+        this() @nogc
+        {
+            if (!initCryptGenRandom(hProvider))
+            {
+                throw defaultAllocator.make!EntropyException("CryptAcquireContextW failed.");
+            }
+            assert(hProvider > 0, "hProvider not properly initialized.");
+        }
+
+        ~this() @nogc nothrow @safe
+        {
+            if (hProvider > 0)
+            {
+                (() @trusted => CryptReleaseContext(hProvider, 0))();
+            }
+        }
+
+        /**
+         * Returns: Minimum bytes required from the entropy source.
+         */
+        override @property ubyte threshold() const pure nothrow @safe @nogc
+        {
+            return 32;
+        }
+
+        /**
+         * Returns: Whether this entropy source is strong.
+         */
+        override @property bool strong() const pure nothrow @safe @nogc
+        {
+            return true;
+        }
+
+        /**
+         * Poll the entropy source.
+         *
+         * Params:
+         *  output = Buffer to save the generate random sequence (the method will
+         *           to fill the buffer).
+         *
+         * Returns: Number of bytes that were copied to the $(D_PARAM output)
+         *          or $(D_PSYMBOL Nullable!ubyte.init) on error.
+         */
+        override Nullable!ubyte poll(out ubyte[maxGather] output) @nogc nothrow @safe
+        in
+        {
+            assert(hProvider > 0, "hProvider not properly initialized.");
+        }
+        do
+        {
+            Nullable!ubyte ret;
+            if ((() @trusted => CryptGenRandom(hProvider, output.length, cast(PBYTE) output.ptr))())
+            {
+                ret = cast(ubyte) (output.length);
+            }
+            return ret;
+        }
+    }
+
+    private unittest
```

Comment:
Please add attributes to the unittest and `private` can go (my own code in this module is pretty old, so sorry for the wrong example):
```d
@nogc @system unittest
{
    ...
}
```